pulsar/internal/host_resolve.go

```diff
@@ -0,0 +1,107 @@
+package internal
+
+import (
+	"errors"
+	"math/rand"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// HostResolve is used to implement multihost
+type HostResolve interface {
+	GetServiceURL() string
+	GetHost() (*url.URL, error)
+	GetHostURL() (string, error)
+	ResolveHost() (*url.URL, error)
+}
+
+type hostResolve struct {
+	serviceURL   string
+	Host         []*url.URL
+	CurrentIndex int
+}
+
+func NewHostResolve(serviceURL string) (HostResolve, error) {
+	h := &hostResolve{
+		serviceURL: serviceURL,
+	}
+
+	// resolve host
+	hosts := strings.Split(h.serviceURL, ",")
+	if len(hosts) == 0 || !checkPrefix(hosts[0]) {
+		return nil, errors.New("invalid service URL")
+	}
+
+	for k := range hosts {
+		if !checkPrefix(hosts[k]) {
```

Comment:
Maybe `hosts[k - 1]` is a better choice than `hosts[0]`?

For example, `"pulsar://host1,pulsar+ssl://host2,host3"`, in which I think `hosts3` is expected to be `pular+ssl` rather than `pulsar`

Reply:
Oh, This is something I didn’t think about!